py/src/braintrust/oai.py

```diff
@@ -1,41 +1,128 @@
-# TODO MOVE TO AUTOEVALS IMPORT
-# https://github.com/braintrustdata/braintrust/issues/218
+import time
 
-import json
-import sqlite3
+from .logger import current_span
 
-from .cache import CACHE_PATH
 
-OAI_CACHE = CACHE_PATH / "oai.sqlite"
+class ChatCompletionWrapper:
+    def __init__(self, chat):
+        self.chat = chat
 
-_CONN = None
+    def create(self, *args, **kwargs):
+        params = self._parse_params(kwargs)
+        stream = kwargs.get("stream", False)
 
+        span = current_span().start_span(name="OpenAI Chat Completion", **params)
+        should_end = True
+        try:
+            start = time.time()
+            response = self.chat.create(*args, **kwargs)
+            if stream:
 
-def open_cache():
-    global _CONN
-    if _CONN is None:
-        _CONN = sqlite3.connect(OAI_CACHE)
-        _CONN.execute("CREATE TABLE IF NOT EXISTS cache (params text, response text)")
-    return _CONN
+                def gen():
+                    try:
+                        first = True
+                        all_results = []
+                        for item in response:
+                            if first:
+                                span.log(
+                                    metrics={
+                                        "time_to_first_token": time.time() - start,
+                                    }
+                                )
+                                first = False
+                            all_results.append(item)
+                            yield item
+                        span.log(output=all_results)
+                    finally:
+                        span.end()
 
+                should_end = False
+                return (x for x in gen())
```

Comment:
does `return gen()` not work for some reason?

Reply:
It does